keras_nlp/utils/text_generation.py

```diff
@@ -48,7 +37,42 @@ def validate_token_probability_fn(token_probability_fn, prompt):
         )
 
 
-def mask_tokens_after_end_token(prompt, max_length, end_token_id, pad_token_id):
+def _pad_prompts(prompt, max_length, pad_token_id):
+    # Maintain a mask tensor of shape `(batch_size,max_length)` which will
+    # control the updates at every step.  At the same time, pad the prompt
+    # with `pad_token_id` to `max_length`.
+    if isinstance(prompt, tf.Tensor):
+        shape = tf.shape(prompt)
+        batch_size = shape[0]
+        length = shape[1]
+
+        mask = tf.ones_like(prompt, dtype=tf.bool)
+        mask_padding = tf.fill((batch_size, max_length - length), False)
+        mask = tf.concat([mask, mask_padding], axis=1)
+
+        padding = tf.fill((batch_size, max_length - length), pad_token_id)
+        prompt = tf.concat((prompt, padding), axis=1)
+    elif isinstance(prompt, tf.RaggedTensor):
+        batch_size = prompt.nrows()
+        length = tf.math.reduce_min(tf.RaggedTensor.row_lengths(prompt))
```

Comment:
Could this use bounding shape?
https://www.tensorflow.org/api_docs/python/tf/RaggedTensor#bounding_shape

Reply:
Heh, right.